library/Frontend/src/Juvix/Frontend/Parser.hs

```diff
@@ -449,6 +451,64 @@ nameParser =
   P.try (P.skipSome (P.char J.hash) *> fmap Types.Implicit prefixSymbol)
     <|> Types.Concrete <$> prefixSymbol
 
+--------------------------------------------------
+-- Effect handler parser
+--------------------------------------------------
+
+handlerParser :: Parser Types.Handler
+handlerParser = do
+  reserved "handler"
+  name <- prefixSymbolSN
+  reserved "where"
+  ops <- P.sepEndBy opParser (skipLiner J.comma)
+  ret <- retParser
+  pure $ Types.Hand name ops ret
```

Comment:
This seems rather inflexible as you have to include the `return` parser as the last parser, otherwise we get an error.

Reply:
I do agree, the order shouldn't be inflexible.

Reply:
Do we need to consider `return` part of the syntax, or could it just be a regular function? I guess for parsing it shouldn't matter, the lack of a `return` I guess is a something that could/should be checked later on in the compilation process? Depends on how we look at these things.